test/unittests/client/toolchain/autotools/autotools_toolchain_test.py

```diff
@@ -605,3 +605,29 @@ def test_conf_compiler_executable():
     be = AutotoolsToolchain(conanfile)
     env = be.vars()
     assert "/c/my/path/myg++" == env["CXX"]
+
+
+def test_autotools_toolchain_conf_configure_args():
+    """Validate that tools.gnu:configure_args are passed to configure command only.
+
+       The configure args should be passed as list only.
+    """
+    f = temp_folder()
+    os.chdir(f)
+    conanfile = ConanFileMock()
+    conanfile.settings = MockSettings({"os": "Linux", "arch": "x86_64"})
+    conanfile.conf = Conf()
+    conanfile.conf.define("tools.gnu:configure_args", ["--foo", "--bar"])
+
+    be = AutotoolsToolchain(conanfile)
+    be.generate_args()
+    obj = load_toolchain_args()
+    assert "--foo --bar" in obj["configure_args"]
+    # make sure it does not forward to make
+    assert "--foo" not in obj["make_args"]
+
+    conanfile.conf.define("tools.gnu:configure_args", "--foo --bar")
+    with pytest.raises(ConanException) as expected:
+        AutotoolsToolchain(conanfile)
+    assert "[conf] tools.gnu:configure_args must be a list-like object. "\
+           "The value '--foo --bar' introduced is a str object" in str(expected)
```

Comment:
I think that we can remove this check as it's part of the `conf.get(..., check_type="")` behavior:

```suggestion
```